src/riscv/lib/src/jit/state_access.rs

```diff
@@ -522,41 +501,17 @@ impl<'a, MC: MemoryConfig> JsaCalls<'a, MC> {
         let exception_slot = self.exception_ptr_slot(builder);
         let exception_ptr = exception_slot.ptr(builder);
 
-        let memory_store = match V::WIDTH {
-            LoadStoreWidth::Byte => self.memory_store_u8.get_or_insert_with(|| {
-                self.module
-                    .declare_func_in_func(self.imports.memory_store_u8, builder.func)
-            }),
-            LoadStoreWidth::Half => self.memory_store_u16.get_or_insert_with(|| {
-                self.module
-                    .declare_func_in_func(self.imports.memory_store_u16, builder.func)
-            }),
-            LoadStoreWidth::Word => self.memory_store_u32.get_or_insert_with(|| {
-                self.module
-                    .declare_func_in_func(self.imports.memory_store_u32, builder.func)
-            }),
-            LoadStoreWidth::Double => self.memory_store_u64.get_or_insert_with(|| {
-                self.module
-                    .declare_func_in_func(self.imports.memory_store_u64, builder.func)
-            }),
-        };
-
-        let value = match V::WIDTH {
-            LoadStoreWidth::Byte | LoadStoreWidth::Half | LoadStoreWidth::Word => {
-                builder.ins().ireduce(V::IR_TYPE, value.to_value())
-            }
-            LoadStoreWidth::Double => value.to_value(),
-        };
+        let value = V::from_xvalue_ir(builder, value);
 
-        let call = builder.ins().call(*memory_store, &[
-            core_ptr.to_value(),
-            phys_address.to_value(),
+        let is_exception = ext_calls::call4(
+            &self.target_config,
+            builder,
+            self::memory_store,
+            unsafe { core_ptr.as_mut() },
```

Comment:
A safety comment is missing. Same below.

Reply:
Thanks! Added [here](https://github.com/tezos/riscv-pvm/compare/c77ac9b18e054e0c9a6151d243979426d91c201b..2a9514d6c89db3ecac19d705b62f6f8b9b55fecc).